packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -104,266 +68,583 @@ describe("Garbage Collection Tests", () => {
-        assert(
-            mockLogger.matchEvents([
-                { eventName: inactiveObjectRevivedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
-            ]),
-            "inactiveObjectRevived event not generated as expected",
-        );
+        it("generates inactive events for nodes that are inactive on load - snapshot is in old format", async () => {
+            // Create GC details for node 3's GC blob whose unreferenced time was > deleteTimeoutMs ago.
+            // This means this node should become inactive as soon as its data is loaded.
+            const node3GCDetails: IGarbageCollectionSummaryDetails = {
+                gcData: { gcNodes: { "/": [] } },
+                unrefTimestamp: Date.now() - (deleteTimeoutMs + 100),
+            };
+            const node3Snapshot = getDummySnapshotTree();
+            node3Snapshot.blobs[gcBlobKey] = "node3GCDetails";
+
+            // Create a base snapshot that contains snapshot tree of node 3.
+            const baseSnapshot = getDummySnapshotTree();
+            baseSnapshot.trees[nodes[3].slice(1)] = node3Snapshot;
+
+            // Set up the getNodeGCDetails function to return the GC details for node 3 when asked by garbage collector.
+            const getNodeGCDetails = (blobId: string) => {
+                if (blobId === "node3GCDetails") {
+                    return node3GCDetails;
+                }
+                return {};
+            };
+            const garbageCollector = createGarbageCollector(baseSnapshot, getNodeGCDetails);
+
+            // Remove node 3's reference from node 2 so that it is still unreferenced. The GC details from the base
+            // summary is not loaded until the first time GC is run, so do that immediately.
+            defaultGCData.gcNodes[nodes[2]] = [];
+            await garbageCollector.collectGarbage({ runGC: true });
+
+            // Change node 3. This should result in an inactiveObjectChanged event for it since it should be inactive.
+            garbageCollector.nodeChanged(nodes[3]);
+            assert(
+                mockLogger.matchEvents([
+                    { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
+                ]),
+                "inactiveObjectChanged event not generated as expected",
+            );
+
+            // Add a reference to node 3 from node 2. Run GC and validate that we get inactiveObjectRevived for node 3.
+            defaultGCData.gcNodes[nodes[2]] = [ nodes[3] ];
+            await garbageCollector.collectGarbage({ runGC: true });
+            assert(
+                mockLogger.matchEvents([
+                    { eventName: inactiveObjectRevivedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
+                ]),
+                "inactiveObjectRevived event not generated as expected",
+            );
+        });
+
+        it(`generates inactive events for nodes that are inactive on load - GC state is present in multiple` +
```

Comment:
Consider making names shorter, and instead putting more detailed description in comment above it.